### PR TITLE
ignore-tree-if-no-tree

### PR DESCRIPTION
**Pull request by antgonza**

### CODE REVIEW

**Comment by antgonza** [comment]
Solves https://github.com/biocore/qiita/issues/3039

File: qp_qiime2/qp_qiime2.py [hunk 1]
```diff
@@ -342,14 +342,14 @@ def call_qiime2(qclient, job_id, parameters, out_dir):
                 return False, None, ('Error generating taxonomy. Are you '
                                      'sure this artifact has taxonomy?')
             q2params['taxonomy'] = qza
-        else:
+        elif fpath is not None:
```
Comment (ElDeveloper) [inline qp_qiime2/qp_qiime2.py:345]: Can you add an `else` block and add a comment or error explaining what happens when this is the case? Does a None represent a Tree?

**Review by ElDeveloper — CHANGES_REQUESTED** [review]

File: qp_qiime2/qp_qiime2.py [hunk 2]
```diff
@@ -352,6 +352,15 @@ def call_qiime2(qclient, job_id, parameters, out_dir):
             elif exists(fpath):
                 qza = qiime2.Artifact.load(fpath)
             q2params[k] = qza
+        else:
+            # adding an else for completeness: if we get here then we should
+            # ignore that parameter/input passed. By design, this should only
+            # happen in one scenario: the user selected an artifact, in
+            # specific a tree, that doesn't exist. This was added while solving
+            # https://github.com/biocore/qiita/issues/3039. However, in the
+            # future in might be useful to always ignore anything that doesn't
```
Comment (ElDeveloper) [inline qp_qiime2/qp_qiime2.py]:
```suggestion
            # future it might be useful to always ignore anything that doesn't
```

**Review by ElDeveloper — APPROVED** [review]

**Comment by ElDeveloper** [comment]
Thanks @antgonza

**Pull request merged**